### PR TITLE
Remove vite-plugin-svelte-kit when detected

### PR DESCRIPTION
**Pull request by JReinhold**
Issue: #19280

## What I did

- Remove `vite-plugin-svelte-kit` when detected as per discussions in #19280
- Re-enable SvelteKit repros

## How to test

- Create the `svelte-kit/skeleton-js` sandbox
- Ensure that it both runs in dev and it builds

### CODE REVIEW

File: code/lib/builder-vite/src/utils/without-vite-plugins.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { PluginOption } from 'vite';
+
+// recursively remove all plugins with the given names
+export const withoutVitePlugins = (
+  plugins: PluginOption[] = [],
+  namesToRemove: string[]
```
Comment (IanVS) [inline code/lib/builder-vite/src/utils/without-vite-plugins.ts:6]: I think you've unintentionally helped me add another feature which I've had in mind, which is allowing users to remove plugins from their vite config by providing a list of names, similar to what Histoire allows: https://histoire.dev/guide/config.html#ignoring-plugins.  Having this function will come in handy.

**Review by IanVS — APPROVED** [review]

File: code/lib/builder-vite/src/utils/without-vite-plugins.ts (hunk 1 above)
Comment (JReinhold) [reply code/lib/builder-vite/src/utils/without-vite-plugins.ts:6]: Ha great. I just thought that I'd written it 3 times now that it probably deserves it's own utility.

File: .circleci/config.yml [hunk 2]
```diff
@@ -375,7 +375,7 @@ jobs:
     executor:
       class: medium+
       name: sb_playwright
-    parallelism: 18
+    parallelism: 22
```
Comment (JReinhold) [inline .circleci/config.yml:378]:
```suggestion
    parallelism: 20
```

**Review by shilman — COMMENTED** [review]
@JReinhold FWIW I think this is a `maintenance` PR since it touches user-facing code in a significant way. `build` means it's completely internal to our build system.

**Comment by JReinhold** [comment]
@shilman We really need a written guide on those labels soon. 😬

**Comment by gbkwiatt** [comment]
Can someone explain how it should work now ? Previously I've had workaround `const plugins = config.plugins.flat(1).filter((p) => p.name !== 'vite-plugin-svelte-kit');` but now it doesn't work on `42` version, and it doesn't build correctly anymore.

**Comment by JReinhold** [comment]
> Can someone explain how it should work now ? Previously I've had workaround `const plugins = config.plugins.flat(1).filter((p) => p.name !== 'vite-plugin-svelte-kit');` but now it doesn't work on `42` version, and it doesn't build correctly anymore.

None of that should be necessary anymore as Storybook automatically handles that, so hopefully you should be able to delete that.

If you're still experiencing issues with your specific setup I might be easier to reach out on [Discord](https://discord.gg/storybook). I'm JReinhold#4884

**Pull request merged**